cmake/SetupCompilerOptions.cmake

```diff
@@ -265,7 +265,7 @@ if (BLT_CXX_STD)
         if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "XL")
             message(FATAL_ERROR "XL does not support C++17.")
         endif()
-        if (ENABLE_CUDA)
+        if (ENABLE_CUDA AND (NOT DEFINED CMAKE_CUDA_COMPILE_FEATURES OR (NOT "cuda_std_17" IN_LIST CMAKE_CUDA_COMPILE_FEATURES)))
```

Comment:
Thanks @evaleev!

This looks reasonable to me, but we should probably add a host-config to test this.
Thoughts: @white238 @davidbeckingsale  ?

Reply:
Thanks! at the time there was no support for CUDA + C++17 and we never revisited.  I appreciate fixing this.

As for the host-configs, we need a general update of BLT's host-configs and will create an issue so we dont forget.

Reply:
#388